src/sonic-py-common/sonic_py_common/multi_asic.py

```diff
@@ -451,3 +431,41 @@ def validate_namespace(namespace):
         return True
     else:
         return False
+
+
+"""
+One time module initialization. Get all the namespaces from CONFIG_DB.
+get_all_namespaces() gets called lot many times(eg. 'portstat -s all')
+especially when a platform has lot many interfaces across multiple namespaces
+so doing DB connect/close at is time consuming, hence we store this static
+information for the lifetime of the process. This prevents two issues we have seen:
+1. FD leak due to too many files opened by the process
+2. Race condition within multiple threads of the same process
+"""
+front_ns = []
+back_ns = []
+num_asics = get_num_asics()
+if is_multi_asic():
+    for asic in range(num_asics):
+        namespace = "{}{}".format(ASIC_NAME_PREFIX, asic)
+        command = 'sonic-db-cli -n {} CONFIG_DB HGET "DEVICE_METADATA|localhost" "sub_role"'.format(namespace)
```

Comment:
with this change, database container is failing to boot with the error at
"sonic-db-cli -n {} CONFIG_DB HGET "DEVICE_METADATA|localhost" "sub_role""

/usr/local/lib/python3.9/dist-packages/supervisor/options.py:473: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (inc.
self.warnings.warn(
Traceback (most recent call last):
File "/usr/local/bin/sonic-cfggen", line 32, in
from minigraph import minigraph_encoder, parse_xml, parse_device_desc_xml, parse_asic_sub_role, parse_asic_switch_type
File "/usr/local/lib/python3.9/dist-packages/minigraph.py", line 14, in
from portconfig import get_port_config
File "/usr/local/lib/python3.9/dist-packages/portconfig.py", line 10, in
from sonic_py_common.multi_asic import get_asic_id_from_name
File "/usr/local/lib/python3.9/dist-packages/sonic_py_common/multi_asic.py", line 447, in
raise RuntimeError(
RuntimeError: Command sonic-db-cli -n asic0 CONFIG_DB HGET "DEVICE_METADATA|localhost" "sub_role" failed with stderr None
/usr/local/lib/python3.9/dist-packages/supervisor/options.py:473: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (inc.
self.warnings.warn(
Unlinking stale socket /var/run/supervisor.sock